### ISSUE TITLE
2.1.1 - Linux fonts not rendering correctly

### ISSUE THREAD

**Issue by SamVerschueren** [ariya/phantomjs#14013]
I noticed that fonts on Linux aren't rendered correctly. The font used in `todomvc.com` is `Helvetica` with an `Arial` fallback. It renders well on phantom 1.9 but doesn't work on phantom 2.x.
#### PhantomJS 2.1.1

Font is definitely not `Helvetica`.

![screen shot 2016-02-17 at 09 23 48](https://cloud.githubusercontent.com/assets/1913805/13103659/357d975a-d558-11e5-903a-4209c8d60238.png)
#### PhantomJS 1.9.x

Font `Helvetica` is correct.

![screen shot 2016-02-17 at 09 25 56](https://cloud.githubusercontent.com/assets/1913805/13103706/7ba81566-d558-11e5-9059-dcddea8acb76.png)

**Comment by 4ndv** [comment]
Having same issue with rendering on linux.

Same medium.com article, rendered by mac and linux:

Mac (pretty ok):
![image](https://cloud.githubusercontent.com/assets/10115449/13549438/5b4f3f26-e32f-11e5-9df5-e30518cf4fab.png)

Linux (my eyes are bleeding):
![image](https://cloud.githubusercontent.com/assets/10115449/13549442/6a8dd088-e32f-11e5-8718-f0bce376c6b5.png)

**Comment by bmustata** [comment]
Some fonts don't work properly under Linux.

**Comment by jasonswearingen** [comment]
i am also seeing the issue.  it looks like phantomjs is choosing a fallback font even though other better options exist.

**Comment by bmustata** [comment]
Troubleshooting the fonts issue is currently very hard on linux/unix machines.

**Comment by blimmer** [comment]
I was able to get helvetica TTF fonts working on Ubuntu - instructions [here](https://benlimmer.com/2016/08/13/using-ttf-fonts-with-phantom-js-ubuntu/).

**Comment by JKring** [comment]
Having the same issue. Is @blimmer's approach the recommended workaround?

**Comment by jasonswearingen** [comment]
@JKring  i have had the same problems, it looks like there are actually multiple issues at hand here.

Following those instructions from @blimmer  help solve a number of font issues, but others persist.  (fonts that just will not render properly on linux regardless of the font being installed)

**Comment by JKring** [comment]
@jasonswearingen Yea, I had some luck with @blimmer's approach, but at least one font (served only as a `.woff`) is still not rendering correctly. Unfortunately, I'm testing on behalf of clients, so I don't have control over how the fonts are served. Makes debugging tough.

**Comment by terwey** [comment]
I am currently using [svg2png](https://github.com/domenic/svg2png) which uses PhantomJS as the underlying rendering engine. I ran into the same problem.

I solved it by defining the font inside the SVG directly:

```
 <defs>
  <style type="text/css">@import url(/usr/share/fonts/truetype/msttcorefonts/verdana.ttf);</style>
 </defs>
```

However since I do not have a fully functional installation of PhantomJS at hand to throw a file at I cannot confirm if this technique also works for normal websites. However it is worth a try.
